fivox/densityImageSource.hxx

```diff
@@ -0,0 +1,82 @@
+
+/* Copyright (c) 2014-2016, EPFL/Blue Brain Project
+ *                          Daniel.Nachbaur@epfl.ch
+ *
+ * This file is part of Fivox <https://github.com/BlueBrain/Fivox>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef FIVOX_DENSITYIMAGESOURCE_HXX
+#define FIVOX_DENSITYIMAGESOURCE_HXX
+
+#include "densityImageSource.h"
+
+#include <itkProgressReporter.h>
+
+namespace fivox
+{
+
+template< typename TImage > DensityImageSource< TImage >::DensityImageSource()
+    : ImageSource< TImage >()
+{
+}
+
+template< typename TImage >
+void DensityImageSource< TImage >::GenerateData()
+{
+    Superclass::_progressObserver->reset();
+
+    auto image = Superclass::GetOutput();
+    image->Allocate();
+    image->FillBuffer( 0 );
+
+    auto source = Superclass::_eventSource;
+    itk::ProgressReporter progress( this, 0, source->getNumChunks( ));
+    size_t totalEvents = 0;
+    for( size_t i = 0; i < source->getNumChunks(); ++i )
+    {
+        totalEvents += source->load( i, 1 );
+
+        const float* __restrict__ posx = source->getPositionsX();
+        const float* __restrict__ posy = source->getPositionsY();
+        const float* __restrict__ posz = source->getPositionsZ();
+        const float* __restrict__ values = source->getValues();
+
+        #pragma vector aligned
+        for( size_t j = 0; j < source->getNumEvents(); ++j )
```

Comment:
Where would the region come from? It's iterating over events which are in a region that is not known a priori. Besides, right now I'm not seeing any optimization potential here as VTune just shows me grey bars for load() :)